code/game/objects/structures/electricchair.dm

```diff
@@ -1,52 +1,132 @@
+#define ELECTRIC_CHAIR_WAIT_TIME 5
+
 /obj/structure/chair/e_chair
 	name = "electric chair"
 	desc = "Looks absolutely SHOCKING!"
 	icon_state = "echair0"
-	var/obj/item/assembly/shock_kit/part = null
 	var/last_time = 1
 	item_chair = null
 
 /obj/structure/chair/e_chair/Initialize()
 	. = ..()
-	add_overlay(mutable_appearance('icons/obj/chairs.dmi', "echair_over", MOB_LAYER + 1))
+	if(!stored_kit)
+		stored_kit = new(src)
+		stored_kit.master = src
+		AddComponent(/datum/component/electrified_chair, stored_kit)
 
 /obj/structure/chair/e_chair/attackby(obj/item/W, mob/user, params)
 	if(W.tool_behaviour == TOOL_WRENCH)
 		var/obj/structure/chair/C = new /obj/structure/chair(loc)
 		W.play_tool_sound(src)
 		C.setDir(dir)
-		part.forceMove(loc)
-		part.master = null
-		part = null
+		stored_kit.forceMove(loc)
+		stored_kit.master = null
+		stored_kit = null
 		qdel(src)
 
-/obj/structure/chair/e_chair/proc/shock()
-	if(last_time + 50 > world.time)
-		return
-	last_time = world.time
 
-	// special power handling
-	var/area/A = get_area(src)
-	if(!isarea(A))
-		return
-	if(!A.powered(AREA_USAGE_EQUIP))
+/datum/component/electrified_chair
+	///the chair that we are attached to
+	var/obj/structure/chair/parent_chair
+	///the shock kit attached to parent_chair
+	var/obj/item/assembly/shock_kit/used_shock_kit
+	///the mob buckled to parent_chair, if any
+	var/mob/living/guinea_pig
+	///the guinea pig can only be shocked if this reaches 5 seconds
+	var/time_since_last_shock = 0
+	///this is casted to the overlay we put on parent_chair
+	var/mutable_appearance/shock_helmet_overlay
+	///the original name of parent_chair
+	var/original_name
+
+/datum/component/electrified_chair/Initialize(obj/item/assembly/shock_kit/input_shock_kit)
+	if(!istype(parent, /obj/structure/chair) || !istype(input_shock_kit, /obj/item/assembly/shock_kit))
+		return COMPONENT_INCOMPATIBLE
+	parent_chair = parent
+	used_shock_kit = input_shock_kit
+	RegisterSignal(parent_chair, COMSIG_PARENT_PREQDELETED, .proc/delete_self)
+	RegisterSignal(used_shock_kit, COMSIG_PARENT_PREQDELETED, .proc/delete_self)
+
+	RegisterSignal(parent_chair, COMSIG_MOVABLE_BUCKLE, .proc/on_buckle)
+	RegisterSignal(parent_chair, COMSIG_ATOM_EXIT, .proc/check_shock_kit)
+	shock_helmet_overlay = image('icons/obj/chairs.dmi', "echair_over", OBJ_LAYER)
+	parent_chair.add_overlay(shock_helmet_overlay)
+	original_name = parent_chair.name
+	parent_chair.name = "electrified [original_name]"
+
+	if(parent_chair.has_buckled_mobs())
+		for(var/m in parent_chair.buckled_mobs)
+			var/mob/living/possible_guinea_pig = m
+			if(on_buckle(src, possible_guinea_pig, FALSE))
```

Comment:
Named argument for `FALSE`, but it doesn't look like you need it anyway.

Reply:
now gone